impeller/renderer/blit_command.h

```diff
@@ -0,0 +1,30 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
+
+#include <variant>
+
+#include "impeller/renderer/texture.h"
+
+namespace impeller {
+
+struct BlitCommand {
```

Comment:
Let's just use inheritance instead please. This is going to get out of hand once we add variants for texture-buffer copies, indirect buffers, etc..

Reply:
This will also make the blit-pass deal with just one AddCommand variant. We can figure out what each command does in its its subclass.

Reply:
This is my aggressive preference for contiguous memory perhaps getting the best of me. Note that this just the struct for encoding commands. Does the BlitPass API itself look OK to you, or would you rather we expose the BlitCommand hierarchy as part of the API?

Reply:
The BlitPass itself looks great to me.

Reply:
Switched to a class hierarchy.